stdlib/public/stubs/Stubs.cpp

```diff
@@ -151,9 +162,25 @@ static uint64_t swift_floatingPointToString(char *Buffer, size_t BufferLength,
     Precision = std::numeric_limits<T>::max_digits10;
   }
   
+#if defined(__CYGWIN__)
+  // Cygwin does not support uselocale(), but we can use the locale feature 
+  // in stringstrem object.
+  std::ostringstream ValueStream;
+  ValueStream.width(0);
+  ValueStream.precision(Precision);
+  ValueStream.imbue(std::locale::classic());
+  ValueStream << Value;
+  std::string  ValueString(ValueStream.str());
+  int i = ValueString.length();
```

Comment:
The variable should be declared as `size_t` instead of a cast on the line below.